src/Processors/Formats/Impl/ArrowColumnToCHColumn.cpp

```diff
@@ -277,8 +280,16 @@ static ColumnWithTypeAndName readColumnWithDate32Data(std::shared_ptr<arrow::Chu
             {
                 Int32 days_num = static_cast<Int32>(chunk.Value(value_i));
                 if (days_num > DATE_LUT_MAX_EXTEND_DAY_NUM || days_num < -DAYNUM_OFFSET_EPOCH)
-                    throw Exception{ErrorCodes::VALUE_IS_OUT_OF_RANGE_OF_DATA_TYPE,
-                            "Input value {} of a column \"{}\" is out of allowed Date32 range, which is [{}, {}]", days_num, column_name, DAYNUM_OFFSET_EPOCH, DATE_LUT_MAX_EXTEND_DAY_NUM};
+                {
+                    if (date_time_overflow_mode == FormatSettings::DateTimeOverflowMode::Saturate)
```

Comment:
1. Sorry, I know I asked before but should we replace the `if` by a `switch`?

2. If the mode is `ignore`, we'll throw an exception. That doesn't sound right.

Reply:
1. Yes, here it is fine to replace it